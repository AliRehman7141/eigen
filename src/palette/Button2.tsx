import React, { useEffect, useImperativeHandle, useRef, useState } from "react"
import { ActivityIndicator, Animated, Easing, View } from "react-native"
import { Text, TouchableWithScale } from "./elements"
import { color } from "./helpers"

type Variant = "primary" | "secondary"

function usePrevious<T>(value: T) {
  const ref = useRef<T>(value)
  useEffect(() => {
    ref.current = value
  }, [value])
  return ref.current
}

function useAnimatedColor(nextColor: string) {
  const nextDrivingValue = useRef(1)
  const driver = useRef(new Animated.Value(0)).current
  const [interpolationConfig, setInterpolationConfig] = useState({
    inputRange: [-1, 0],
    outputRange: [nextColor, nextColor],
  })
  const prevColor = usePrevious(nextColor)

  useEffect(() => {
    if (nextColor !== prevColor) {
      setInterpolationConfig(({ inputRange, outputRange }) => ({
        inputRange: [...inputRange, nextDrivingValue.current],
        outputRange: [...outputRange, nextColor],
      }))
      setTimeout(() => {
        Animated.timing(driver, {
          easing: Easing.ease,
          duration: 100,
          useNativeDriver: false,
          toValue: nextDrivingValue.current,
        }).start()
        nextDrivingValue.current++
      }, 20)
    }
  }, [nextColor])

  return driver.interpolate(interpolationConfig)
}

export const Button2: React.FC<{
  loading: boolean
  states: { [key: string]: { variant: Variant; label: string } }
  currentState: string
  onPress(): void
}> = (props) => {
  const prevProps = usePrevious(props)
  const variantRefs = useRef<Record<string, ButtonVariant | null>>({}).current

  useEffect(() => {
    if (props.currentState !== prevProps.currentState) {
      variantRefs[prevProps.currentState]?.disappear()
      variantRefs[props.currentState]?.appear()
    } else if (props.loading !== prevProps.loading) {
      variantRefs[props.currentState]?.setLoading(props.loading, true)
    }
  }, [props])

  const [isActive, setIsActive] = useState(false)

  return (
    <TouchableWithScale
      disabled={props.loading}
      onPress={() => {
        props.onPress?.()
      }}
      onPressIn={() => {
        setIsActive(true)
      }}
      onPressOut={() => {
        setIsActive(false)
      }}
      style={{ height: 40 }}
    >
      {Object.keys(props.states)
        .filter((k) => k !== props.currentState)
        .concat([props.currentState])
        .map((k) => {
          return (
            <ButtonVariant
              ref={(ref) => {
                variantRefs[k] = ref
              }}
              key={k}
              variant={props.states[k].variant}
              isLoadingAtStart={k === props.currentState ? props.loading : false}
              isVisibleAtStart={k === props.currentState}
              isActive={k === props.currentState && isActive}
            >
              {props.states[k].label}
            </ButtonVariant>
          )
        })}
    </TouchableWithScale>
  )
}

interface ButtonVariant {
  appear(): void
  disappear(): void
  setLoading(loading: boolean, animated: boolean): void
}
const ButtonVariant = React.forwardRef<
  ButtonVariant,
  React.PropsWithChildren<{
    isVisibleAtStart: boolean
    isLoadingAtStart: boolean
    variant: Variant
    isActive: boolean
  }>
>(({ variant, children, isVisibleAtStart, isLoadingAtStart, isActive }, ref) => {
  const opacity = useRef(new Animated.Value(isVisibleAtStart ? 1 : 0)).current
  const scale = useRef(new Animated.Value(1)).current
  const translateY = useRef(new Animated.Value(0)).current

  const { textColor, borderColor, activeBorderColor, backgroundColor, activityIndicatorColor } = getVariantColors(
    variant
  )

  const textOpacity = useRef(new Animated.Value(isLoadingAtStart ? 0 : 1)).current
  const textTranslateY = useRef(new Animated.Value(0)).current

  const activityIndicatorOpacity = useRef(new Animated.Value(isLoadingAtStart ? 1 : 0)).current
  const activityIndicatorTranslateY = useRef(new Animated.Value(0)).current

  const animatedBorderColor = useAnimatedColor(isActive ? activeBorderColor : borderColor)

  useImperativeHandle(
    ref,
    (): ButtonVariant => {
      return {
        appear() {
          this.setLoading(false, false)
          opacity.setValue(0)
          scale.setValue(0.97)
          setTimeout(() => {
            Animated.parallel([
              Animated.timing(opacity, { toValue: 1, useNativeDriver: true, easing: Easing.ease, duration: 130 }),
              Animated.timing(scale, { toValue: 1, useNativeDriver: true, easing: Easing.ease, duration: 130 }),
            ]).start()
          })
        },
        disappear() {
          opacity.setValue(1)
          scale.setValue(1)
          translateY.setValue(0)
          setTimeout(() => {
            Animated.parallel([
              Animated.timing(opacity, { toValue: 0, useNativeDriver: true, easing: Easing.ease, duration: 130 }),
              Animated.timing(scale, { toValue: 0.9, useNativeDriver: true, easing: Easing.ease, duration: 130 }),
            ]).start()
          })
        },
        setLoading(loading, animated) {
          const diff = 15
          const duration = 150
          const easing = Easing.inOut(Easing.ease)
          if (animated) {
            if (loading) {
              textOpacity.setValue(1)
              textTranslateY.setValue(0)
              activityIndicatorOpacity.setValue(0)
              activityIndicatorTranslateY.setValue(-diff)
              setTimeout(() => {
                Animated.parallel([
                  Animated.timing(textOpacity, {
                    toValue: 0,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                  Animated.timing(textTranslateY, {
                    toValue: diff,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                  Animated.timing(activityIndicatorOpacity, {
                    toValue: 1,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                  Animated.timing(activityIndicatorTranslateY, {
                    toValue: 0,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                ]).start()
              })
            } else {
              textOpacity.setValue(0)
              textTranslateY.setValue(-diff)
              activityIndicatorOpacity.setValue(1)
              activityIndicatorTranslateY.setValue(0)

              setTimeout(() => {
                Animated.parallel([
                  Animated.timing(textOpacity, {
                    toValue: 1,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                  Animated.timing(textTranslateY, {
                    toValue: 0,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                  Animated.timing(activityIndicatorOpacity, {
                    toValue: 0,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                  Animated.timing(activityIndicatorTranslateY, {
                    toValue: diff,
                    useNativeDriver: true,
                    easing,
                    duration,
                  }),
                ]).start()
              })
            }
          } else {
            textOpacity.setValue(loading ? 0 : 1)
            textTranslateY.setValue(0)

            activityIndicatorOpacity.setValue(loading ? 1 : 0)
            activityIndicatorTranslateY.setValue(0)
          }
        },
      }
    }
  )

  return (
    <StretchAbsolutely
      style={{
        opacity,
        overflow: "hidden",
        transform: [
          {
            scale,
          },
        ],
      }}
    >
      <StretchAbsolutely
        style={{
          opacity: activityIndicatorOpacity.interpolate({
            inputRange: [0, 1],
            outputRange: [1, 0.7],
          }),
        }}
      >
        <StretchAbsolutely
          style={{
            backgroundColor,
            borderColor: animatedBorderColor,
            borderWidth: 1,
            borderRadius: 2,
            overflow: "hidden",
          }}
        ></StretchAbsolutely>
      </StretchAbsolutely>
      <StretchAbsolutely
        pointerEvents="none"
        style={{
          alignItems: "center",
          justifyContent: "center",
          opacity: textOpacity,
          transform: [
            {
              translateY: textTranslateY,
            },
          ],
        }}
      >
        <Text variant="mediumText" color={textColor} textAlign="center">
          {children}
        </Text>
      </StretchAbsolutely>
      <StretchAbsolutely
        pointerEvents="none"
        style={{
          alignItems: "center",
          justifyContent: "center",
          opacity: activityIndicatorOpacity,
          transform: [
            {
              translateY: activityIndicatorTranslateY,
            },
          ],
        }}
      >
        <ActivityIndicator color={activityIndicatorColor} />
      </StretchAbsolutely>
    </StretchAbsolutely>
  )
})

const StretchAbsolutely: typeof Animated.View = ({ style, ...others }) => {
  return (
    <Animated.View
      style={[
        {
          position: "absolute",
          top: 0,
          left: 0,
          right: 0,
          bottom: 0,
        },
        style,
      ]}
      {...(others as any)}
    />
  )
}

function getVariantColors(
  variant: Variant
): {
  textColor: string
  borderColor: string
  activeBorderColor: string
  backgroundColor: string
  activityIndicatorColor: string
} {
  if (variant === "primary") {
    return {
      activeBorderColor: "black",
      textColor: "white",
      borderColor: "black",
      backgroundColor: "black",
      activityIndicatorColor: "white",
    }
  } else {
    return {
      textColor: "black",
      borderColor: color("black10"),
      activeBorderColor: color("black60"),
      backgroundColor: "white",
      activityIndicatorColor: "black",
    }
  }
}
