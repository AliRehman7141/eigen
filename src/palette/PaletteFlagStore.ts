import AsyncStorage from "@react-native-community/async-storage"
import { zNames } from "lib/utils/zustandPersistNames"
import create from "zustand"
import { persist } from "zustand/middleware"

interface PaletteFlagState {
  allowV3: boolean
  setAllowV3: (value: boolean) => void
}

export const usePaletteFlagStore = create<PaletteFlagState>(
  persist(
    (set) => ({
      allowV3: __TEST__,
      setAllowV3: (value) => set((_state) => ({ allowV3: value })),
    }),
    { name: zNames.paletteFlag, getStorage: () => AsyncStorage }
  )
)
