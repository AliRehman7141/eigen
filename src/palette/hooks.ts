import { useTheme } from "./Theme"

export const useColor = () => useTheme().color
export const useSpace = () => useTheme().space

/** Returns a config specific to the current theme. */
export const useThemeConfig = <T, U>({ v3 }: { v3: U }): U | T => {
  // const { theme = { id: "v2" } } = useTheme()
  return v3
}
