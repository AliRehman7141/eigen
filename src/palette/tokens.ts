export type SpacingUnit = "0.5" | "1" | "2" | "4" | "6" | "12"

const spacing: Record<SpacingUnit, `${number}px`> = {
  // unit: px value
  "0.5": "5px",
  "1": "10px",
  "2": "20px",
  "4": "40px",
  "6": "60px",
  "12": "120px",
}

export const tokens = {
  spacing,
}
