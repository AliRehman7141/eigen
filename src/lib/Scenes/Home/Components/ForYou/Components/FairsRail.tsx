import { Sans, Separator } from "@artsy/palette"
import { FairsRail_fairs_module } from "__generated__/FairsRail_fairs_module.graphql"
import React, { Component } from "react"
import { View } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import styled from "styled-components/native"

import { CardScrollView, CardScrollViewCard } from "lib/Components/Home/CardScrollView"
import ImageView from "lib/Components/OpaqueImageView/OpaqueImageView"
import Switchboard from "lib/NativeModules/SwitchBoard"
import SectionTitle from "lib/Scenes/Home/Components/SectionTitle"

import { concat, take } from "lodash"

const ARTWORKS_HEIGHT = 180

interface Props {
  fairs_module: FairsRail_fairs_module
}

export class FairsRail extends Component<Props, null> {
  render() {
    if (!this.props.fairs_module.results.length) {
      return
    }

    const fairCards = this.props.fairs_module.results.map(result => {
      // Fairs are expected to always have >= 3 artworks. We can make
      // assumptions about this in UI layout, but should still be cautious
      // to avoid crashes if this assumption is broken.
      const artworkImageURLs = take(
        concat(
          [result.heroImage.url],
          result.followedArtistArtworks.edges.map(edge => edge.node.image.url),
          result.otherArtworks.edges.map(edge => edge.node.image.url)
        ),
        3
      )
      return (
        <CardScrollViewCard
          key={result.slug}
          onPress={() => Switchboard.presentNavigationViewController(this, `${result.slug}?entity=fair`)}
        >
          <View>
            <ArtworkImageContainer>
              <ImageView width={ARTWORKS_HEIGHT} height={ARTWORKS_HEIGHT} imageURL={artworkImageURLs[0]} />
              <Division />
              <View>
                <ImageView width={ARTWORKS_HEIGHT / 2} height={ARTWORKS_HEIGHT / 2} imageURL={artworkImageURLs[1]} />
                <Division horizontal />
                <ImageView width={ARTWORKS_HEIGHT / 2} height={ARTWORKS_HEIGHT / 2} imageURL={artworkImageURLs[2]} />
              </View>
            </ArtworkImageContainer>
            <MetadataContainer>
              <Sans numberOfLines={1} weight="medium" size="3t">
                {result.name}
              </Sans>
              <Sans numberOfLines={1} size="3t">
                {result.exhibitionPeriod}
              </Sans>
            </MetadataContainer>
          </View>
        </CardScrollViewCard>
      )
    })

    return (
      <View>
        <Title>
          <SectionTitle>Recommended Art Fairs</SectionTitle>
        </Title>
        <CardScrollView>{fairCards}</CardScrollView>
        <Separator />
      </View>
    )
  }
}

// TODO: Similar to the Title component in ArtistRail, we want to clean these all up at once.
const Title = styled(SectionTitle)`
  margin-left: 20;
`

// Default is a vertical division
export const Division = styled.View<{ horizontal?: boolean }>`
  border: 1px solid white;
  ${({ horizontal }) => (horizontal ? "height" : "width")}: 1px;
`

const ArtworkImageContainer = styled.View`
  width: 100%;
  height: ${ARTWORKS_HEIGHT}px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

const MetadataContainer = styled.View`
  /* 13px on bottom helps the margin feel visually consistent around all sides */
  margin: 15px 15px 13px;
`

export default createFragmentContainer(FairsRail, {
  fairs_module: graphql`
    fragment FairsRail_fairs_module on HomePageFairsModule {
      results {
        id
        slug
        profile {
          slug
        }
        name
        exhibitionPeriod
        heroImage: image {
          url(version: "large")
        }
        followedArtistArtworks: filterArtworksConnection(first: 2, includeArtworksByFollowedArtists: true) {
          edges {
            node {
              image {
                url(version: "large")
              }
            }
          }
        }
        otherArtworks: filterArtworksConnection(first: 2) {
          edges {
            node {
              image {
                url(version: "large")
              }
            }
          }
        }
      }
    }
  `,
})
