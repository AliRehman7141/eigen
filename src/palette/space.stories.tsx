import { storiesOf } from "@storybook/react-native"
import { tailwind } from "lib/utils/tailwind"
import { Box, Spacer, Text } from "palette"
import React from "react"
import { View, ViewProps } from "react-native"
import { withThemeV3 } from "storybook/decorators"
import { List } from "storybook/helpers"
import { useSpace } from "./hooks"
import { SpacingUnitV3 } from "./Theme"

const SpaceLine = ({ space: theSpace }: { space: SpacingUnitV3 }) => {
  const space = useSpace()
  return (
    <View>
      <View
        style={{
          width: space(theSpace),
          borderBottomWidth: 1,
          borderColor: "black",
          marginBottom: 4,
        }}
      />
      <Text color="black">"{theSpace}"</Text>
      <Text color="black">{`${space(theSpace)}px`}</Text>
    </View>
  )
}

const MyView: React.FC<ViewProps & { tw: string }> = ({ tw, style, ...rest }) => (
  <View style={[style, tailwind(tw)]} {...rest} />
)

const MySpacer: React.FC<{ h?: SpacingUnitV3; v?: SpacingUnitV3 }> = ({ h, v }) => (
  <MyView tw={h ? `ml-${h}` : v ? `mb-${v}` : ""} />
)

const SpaceLine2 = ({ space: theSpace }: { space: SpacingUnitV3 }) => {
  const space = useSpace()
  return (
    <View>
      <View style={tailwind("w-12 border-b border-black-100 mb-4")} />
      <MyView tw="w-12 border-b border-black-100 mb-4" />
      <Text color="black">"{theSpace}"</Text>
      <Text color="black">{`${space(theSpace)}px`}</Text>
    </View>
  )
}

storiesOf("Theme/Space", module)
  .addDecorator(withThemeV3)
  .add("spaces", () => (
    <List style={{ marginLeft: 50 }} contentContainerStyle={{ alignItems: "flex-start" }}>
      <SpaceLine space="0.5" />
      <SpaceLine space="1" />
      <SpaceLine space="6" />
      <SpaceLine2 space="6" />
      <View style={tailwind("bg-red-100 h-6 w-6 border border-black-60")} />

      <>
        <Box backgroundColor="red100" width={10} height={10} />
        <Spacer mb={2} />
        <Box backgroundColor="red100" width={10} height={10} />
      </>
      <>
        <MyView tw="bg-green-100 w-1 h-1" />
        <MyView tw="mb-2" />
        <MyView tw="bg-green-100 w-1 h-1" />
      </>
      <>
        <MyView tw="bg-green-100 w-1 h-1" />
        <MySpacer v="2" />
        <MyView tw="bg-green-100 w-1 h-1" />
      </>
    </List>
  ))
