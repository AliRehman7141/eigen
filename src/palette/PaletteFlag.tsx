import { MenuItem } from "lib/Components/MenuItem"
import { Text } from "palette"
import React from "react"
import { Alert } from "react-native"
import { usePaletteFlagStore } from "./PaletteFlagStore"

export const DevTogglePaletteFlag = () => {
  const currentValue = usePaletteFlagStore((state) => state.allowV3)
  const setValue = usePaletteFlagStore((state) => state.setAllowV3)

  const description = "Allow Palette V3"
  const valText = currentValue ? "Yes" : "No"

  return (
    <MenuItem
      title={description}
      onPress={() => {
        Alert.alert(description, undefined, [
          {
            text: currentValue ? "Keep turned ON" : "Turn ON",
            onPress: () => setValue(true),
          },
          {
            text: currentValue ? "Turn OFF" : "Keep turned OFF",
            onPress: () => setValue(false),
          },
        ])
      }}
      value={
        currentValue ? (
          <Text variant="subtitle" color="black100" fontWeight="bold">
            {valText}
          </Text>
        ) : (
          <Text variant="subtitle" color="black100">
            {valText}
          </Text>
        )
      }
    />
  )
}
