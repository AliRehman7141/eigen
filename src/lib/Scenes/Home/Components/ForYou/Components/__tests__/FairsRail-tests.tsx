import { cloneDeep } from "lodash"
import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import FairsRail from "../FairsRail"

import { Theme } from "@artsy/palette"
import { FairsRail_fairs_module } from "__generated__/FairsRail_fairs_module.graphql"

const artworkNode = {
  node: {
    image: { url: "https://example.com/image.jpg" },
  },
}
const fairsModule: Omit<FairsRail_fairs_module, " $refType"> = {
  results: [
    {
      id: "the-fair",
      name: "The Fair",
      slug: "the-fair",
      exhibitionPeriod: "Monday–Friday",
      profile: { slug: "https://neopets.com" },
      heroImage: { url: "https://example.com/hero.jpg" },
      followedArtistArtworks: {
        edges: [artworkNode, artworkNode, artworkNode],
      },
      otherArtworks: {
        edges: [artworkNode, artworkNode, artworkNode],
      },
    },
    {
      id: "the-profileless-fair",
      slug: "the-profileless-fair",
      name: "The Profileless Fair: You Should Not See Me in Snapshots",
      exhibitionPeriod: "Monday–Friday",
      profile: null,
      heroImage: { url: "https://example.com/hero.jpg" },
      followedArtistArtworks: {
        edges: [artworkNode, artworkNode, artworkNode],
      },
      otherArtworks: {
        edges: [artworkNode, artworkNode, artworkNode],
      },
    },
  ],
}

it("looks correct when rendered", () => {
  const tree = renderer
    .create(
      <Theme>
        <FairsRail fairs_module={fairsModule as any} />
      </Theme>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("looks correct when rendered with fairs missing artworks", () => {
  const fairsCopy = cloneDeep(fairsModule) as any // as any to make edges readwrite
  fairsCopy.results.forEach(result => {
    result.followedArtistArtworks.edges = []
    result.otherArtworks.edges = []
  })
  expect(() =>
    renderer
      .create(
        <Theme>
          <FairsRail fairs_module={fairsModule as any} />
        </Theme>
      )
      .toJSON()
  ).not.toThrow()
})
