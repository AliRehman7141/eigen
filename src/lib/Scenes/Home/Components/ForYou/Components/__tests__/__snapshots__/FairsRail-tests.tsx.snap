// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks correct when rendered 1`] = `
<View>
  <Text
    numberOfLines={0}
    style={
      Array [
        Object {
          "marginLeft": 20,
        },
      ]
    }
  >
    <Text
      numberOfLines={0}
      style={
        Object {
          "fontFamily": "AGaramondPro-Regular",
          "fontSize": 30,
          "textAlign": "left",
        }
      }
    >
      Recommended Art Fairs
    </Text>
  </Text>
  <RCTScrollView
    className="sc-bdVaJa wtrcG"
    contentContainerStyle={
      Object {
        "paddingHorizontal": 20,
      }
    }
    horizontal={true}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderColor": "#e5e5e5",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "overflow": "hidden",
              "width": 270,
            },
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 180,
                  "justifyContent": "space-between",
                  "width": "100%",
                },
              ]
            }
          >
            <AROpaqueImageView
              height={180}
              imageURL="https://example.com/hero.jpg"
              width={180}
            />
            <View
              style={
                Array [
                  Object {
                    "borderColor": "white",
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "width": 1,
                  },
                ]
              }
            />
            <View>
              <AROpaqueImageView
                height={90}
                imageURL="https://example.com/image.jpg"
                width={90}
              />
              <View
                horizontal={true}
                style={
                  Array [
                    Object {
                      "borderColor": "white",
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "height": 1,
                    },
                  ]
                }
              />
              <AROpaqueImageView
                height={90}
                imageURL="https://example.com/image.jpg"
                width={90}
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 13,
                  "marginLeft": 15,
                  "marginRight": 15,
                  "marginTop": 15,
                },
              ]
            }
          >
            <Text
              fontFamily="Unica77LL-Medium"
              fontSize="14px"
              lineHeight="20px"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Unica77LL-Medium",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  Object {},
                ]
              }
            >
              The Fair
            </Text>
            <Text
              fontFamily="Unica77LL-Regular"
              fontSize="14px"
              lineHeight="20px"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Unica77LL-Regular",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  Object {},
                ]
              }
            >
              Monday–Friday
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "width": 15,
            },
          ]
        }
        width="15px"
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderColor": "#e5e5e5",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "overflow": "hidden",
              "width": 270,
            },
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 180,
                  "justifyContent": "space-between",
                  "width": "100%",
                },
              ]
            }
          >
            <AROpaqueImageView
              height={180}
              imageURL="https://example.com/hero.jpg"
              width={180}
            />
            <View
              style={
                Array [
                  Object {
                    "borderColor": "white",
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "width": 1,
                  },
                ]
              }
            />
            <View>
              <AROpaqueImageView
                height={90}
                imageURL="https://example.com/image.jpg"
                width={90}
              />
              <View
                horizontal={true}
                style={
                  Array [
                    Object {
                      "borderColor": "white",
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "height": 1,
                    },
                  ]
                }
              />
              <AROpaqueImageView
                height={90}
                imageURL="https://example.com/image.jpg"
                width={90}
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 13,
                  "marginLeft": 15,
                  "marginRight": 15,
                  "marginTop": 15,
                },
              ]
            }
          >
            <Text
              fontFamily="Unica77LL-Medium"
              fontSize="14px"
              lineHeight="20px"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Unica77LL-Medium",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  Object {},
                ]
              }
            >
              The Profileless Fair: You Should Not See Me in Snapshots
            </Text>
            <Text
              fontFamily="Unica77LL-Regular"
              fontSize="14px"
              lineHeight="20px"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Unica77LL-Regular",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  Object {},
                ]
              }
            >
              Monday–Friday
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#E5E5E5",
          "borderStyle": "solid",
          "borderWidth": 1,
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;
